src/servers/src/http/handler.rs

```diff
@@ -46,8 +47,22 @@ pub async fn sql(
     let start = Instant::now();
     let resp = if let Some(sql) = &params.sql {
         let query_ctx = Arc::new(QueryContext::new());
-        if let Some(db) = params.database {
-            query_ctx.set_current_schema(db.as_ref());
+        if let Some(ref db) = params.database {
+            match sql_handler.is_valid_schema(DEFAULT_CATALOG_NAME, db) {
+                Ok(true) => query_ctx.set_current_schema(db),
+                Ok(false) => {
+                    return Json(JsonResponse::with_error(
+                        format!("Database not found: {}", db),
+                        StatusCode::InvalidArguments,
```

Comment:
I go through `StatusCode` and find `UserNotFound` and `TableNotFound`. Maybe we need a "`DatabaseNotFound`" for this?